server/src/test-integration/java/com/thoughtworks/go/server/service/BuildAssignmentServiceIntegrationTest.java

```diff
@@ -702,6 +702,24 @@ public void shouldCancelAScheduledJobInCaseStageHasBeenRenamed() throws Exceptio
         assertThat(allStagesForRenamedPipeline.byName(p1_2.getFirstStage().getName()).getState(), is(StageState.Building));
     }
 
+    @Test
+    public void shouldCancelAllScheduledJobsInCasePipelineHasBeenDeleted() throws Exception {
+        Material hgMaterial = new HgMaterial("url", "folder");
+        String[] hgRevs = new String[]{"h1"};
+        u.checkinInOrder(hgMaterial, hgRevs);
+
+        ScheduleTestUtil.AddedPipeline p1 = u.saveConfigWith("PIPELINE_WHICH_WILL_EVENTUALLY_WILL_BE_DELETED", u.m(hgMaterial));
+        BuildCause buildCause = BuildCause.createWithModifications(u.mrs(u.mr(u.m(hgMaterial).material, true, hgRevs)), "user");
+        Pipeline originalPipelineRun = scheduleService.schedulePipeline(p1.config.name(), buildCause);
+        pipelineConfigService.deletePipelineConfig(new Username("user"), p1.config, new HttpLocalizedOperationResult());
+
+        CruiseConfig cruiseConfig = configHelper.load();
+        buildAssignmentService.onTimer();   // To Reload Job Plans
+        buildAssignmentService.onConfigChange(cruiseConfig);
```

Comment:
`buildAssignmentService.onConfigChange` would have already taken care of deletion of jobPlans that once belonged to a deleted pipeline. The bug is about these not getting removed if pipeline was deleted via the API.  So, I assume this test will pass even if you remove the fix you have made.

Since you have already called `pipelineConfigService.deletePipelineConfig` from this test to simulate the API, which implicitly calls `BuildAssignmentService.pipelineConfigChangedListener` you should just be able to make your assertions right away.  Remove the call to  `buildAssignmentService.onConfigChange(cruiseConfig)`